hazelcast/include/hazelcast/client/proxy/SerializingProxy.h

```diff
@@ -206,11 +275,11 @@ class HAZELCAST_API SerializingProxy
     inline boost::future<boost::optional<entry_view<K, V>>>
     to_object_entry_view(
       boost::future<boost::optional<map::data_entry_view>> data_future,
-      const K& key)
+      K key)
```

Comment:
Let's count how many times the `key` is copied in this commit and the one before:

**Before** (parameter was `const K&`):
1. The parameter is passed as an lvalue reference so no copies there. (0)
2. The parameter is captured by the lambda by-value ([=]), so one copy. (1).
3. Upon completion, the lambda calls entry_view<K, V>'s ctor, which takes the key by-value, so one copy. (1).

So in total, 2 copies.

**Now** (parameter is `K`):
1. Depending on whether the call site uses an lvalue reference or a nonconst rvalue reference, there may be 1 or 0 copies. The way you call this in get_entry_view will make it copy as I explained in the comment, but assuming that you'll correct it, we can put 0 copies. (0)
2. The parameter is captured by the lambda by-value ([key]), so one copy. (1)
3. Upon completion, the lambda calls entry_view<K, V>'s ctor, which takes the key by-value, so one copy. (1).

So in total, still 2 copies. (at best, if don't change how you call to_object_entry_view it's 3)

A few words on how to optimize the number of copies:
* First of all, AFAIA this PR didn't introduce any extra copies compared to master. The [get_entry_view at master](https://github.com/hazelcast/hazelcast-cpp-client/blob/e8abe511b16e39d89e7c771477b62fb11c575f15/hazelcast/include/hazelcast/client/imap.h#L633-L647) does the same 2 copies: (1) capturing the parameter in lambda, (2) constructing an entry_view. So I think this PR should be fine to merge without any optimization.
* The first copy in the lambda capture is not possible to eliminate in C++11, if we want to keep the lambda. C++14's lambda syntax allows capturing by-move (e.g. [key = std::move(key)]) but C++11's is limited in this sense. However, one could change the lambda to a class with `operator()` overloaded (which what lambdas really are) to achieve the C++14 behavior.
* The copy that happens constructing the `entry_view` is easy to eliminate. Do this:
```C++
[key](...) mutable {
     return entry_view(std::move(key), ...);
}
```
The `mutable` keyword is necessary because by default lambdas are const-qualified, which means any reference to captured variables are also const-qualified and calling `std::move` on const reference has no real effect.